### PR TITLE
Fix box header issues and move summary table to top

### LINKED ISSUE #140: Standardize number of # in box's header
I just tried and feel like header 5 would be better HAHA. Header 4 is somehow too big!

_Originally posted by @DennieDan in https://github.com/AY2425S1-CS2103T-F14a-1/tp/pull/135#discussion_r1810660650_

### LINKED ISSUE #173: UG: Move command summary in UG
Move to after quick start

### PR DESCRIPTION
**Pull request by Soumyadip-cmd**
Closes #140 , closes #173 

Let's:
Change box headers to smaller size by using 5# instead of 4#
Move command summary table to the top

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -46,6 +46,21 @@ If you can type fast, UGTeach can get your contact management tasks done **faste
...
+**Clear**  | `clear`
+**Delete** | `delete INDEX`<br> e.g., `delete 3`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS]…​`<br> e.g.,`edit 2 paid/1200.00 owed/0`
+**Find**   | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
+**Pay**   | `pay INDEX hr/HOURS_PAID`<br> e.g., `pay 1 hr/2.5`
+**List**   | `list`
+**Owe**    | `owe INDEX hr/HOUR_OWED`<br> e.g., `owe 1 hr/1.5`
+**Remind**   | `remind`
+**Help**   | `help`
+**Settle** | `settle INDEX amount/AMOUNT`<br> e.g., `settle 1 amount/500.00`
```
Comment (TY1Fan) [inline docs/UserGuide.md:63]: Thanks for shifting the command summary table up!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -158,15 +173,15 @@ Examples:
...
+<box type="important" header="##### Constraints">
 <markdown>
 * <b>DAY</b> must be one of `Monday` `Tuesday` `Wednesday` `Thursday` `Friday` `Saturday` `Sunday`.
 * <b>KEYWORD</b> must be only alphanumeric characters.
 * At least one of the optional fields must be provided.
 </markdown>
 </box>
 
-<box type="tip" header="#### Tips:">
+<box type="tip" header="##### Tips">
```
Comment (TY1Fan) [inline docs/UserGuide.md:184]: Nice catch! Thanks for spotting and standardizing the colon.

**Review by TY1Fan — APPROVED** [review]
LGTM!

**Review by DennieDan — APPROVED** [review]
Looks nice!
I think can merge this already!

**Pull request merged**